File: app/src/main/java/ru/vyaacheslav/suhov/imeit/activites/MainActivity.java
package ru.vyaacheslav.suhov.imeit.activites;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Build;
import android.os.Bundle;
import android.preference.PreferenceManager;
import android.support.annotation.NonNull;
import android.support.design.widget.NavigationView;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v4.view.GravityCompat;
import android.support.v4.widget.DrawerLayout;
import android.support.v7.app.ActionBarDrawerToggle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Toast;

import java.util.Calendar;
import java.util.Objects;

import ru.vyaacheslav.suhov.imeit.R;
import ru.vyaacheslav.suhov.imeit.ftagments.Session;
import ru.vyaacheslav.suhov.imeit.ftagments.TabFragment;
import ru.vyaacheslav.suhov.imeit.ftagments.time.TimeListFragment;
import ru.vyaacheslav.suhov.imeit.maps.MapsFragment;

public class MainActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener {

    public ActionBarDrawerToggle toggle;
    public java.util.Calendar calendar;
    public FragmentTransaction FT;
    private FragmentManager FM;
    private SharedPreferences prefs;
    public String typeTheme, typeWeek, typeGroupe;
    private Intent settingsIntent;
    public Toolbar tb;
    public NavigationView nv;
    private int weekYear = Calendar.getInstance().get(Calendar.WEEK_OF_YEAR);
    private Menu menu;
    private DrawerLayout dl;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        prefs = PreferenceManager.getDefaultSharedPreferences(getApplicationContext());
        typeTheme = prefs.getString(getString(R.string.pref_theme), "");
        typeWeek = prefs.getString(getString(R.string.week_i), "");
        typeGroupe = prefs.getString(getString(R.string.pref_groupe), "");

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            if (Objects.equals(typeTheme, "Светлая")) {
                setTheme(R.style.ThemeWrithe);
            } else {
                setTheme(R.style.ThemeDark);
            }
        }

        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Интент для запуска окна с настройками
        SharedPreferences prefs = getPreferences(MODE_PRIVATE);
        if (prefs.getBoolean("isFirstRun", true)) {
            Intent intent = new Intent(MainActivity.this, SettingsActivity.class);
            startActivity(intent);
        }
        prefs.edit().putBoolean("isFirstRun", false).apply();

        // Стандартная инициализация компонетов
        tb = findViewById(R.id.toolbar);
        nv = findViewById(R.id.shitstuff);
        dl = findViewById(R.id.drawerLayout);

        calendar = Calendar.getInstance();
        setSupportActionBar(tb);
        //ActionBarToggle - Это иконка DrawerLayout
        toggle = new ActionBarDrawerToggle(this, dl, tb, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
        dl.addDrawerListener(toggle);
        toggle.syncState();

        // Инициализация транзакции фрагментов
        FM = getSupportFragmentManager();
        FT = FM.beginTransaction();
        FT.replace(R.id.containerView, new TabFragment()).commit(); // По умолчанию - расписание.

        // Обработка нажатий на пункты меню.
        nv.setNavigationItemSelectedListener(this);

        settingsIntent = new Intent(MainActivity.this, SettingsActivity.class);
        loadName(); // Загрузка имени группы согласно настройкам;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        this.menu = menu;
        getMenuInflater().inflate(R.menu.menu, menu);
        weekIndicator();
        return super.onCreateOptionsMenu(menu);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case R.id.settings:
                startActivity(settingsIntent);
                finish();
                return true;
            case R.id.weekIndicator:
                weekIndicator();
                return true;
            default:
                return super.onOptionsItemSelected(item);
        }
    }

    public void weekIndicator() {

            if (weekYear % 2 == 0) {
                Toast.makeText(getApplicationContext(), "Текущая неделя: Числитель", Toast.LENGTH_SHORT).show();
                menu.getItem(0).setIcon(getResources().getDrawable(R.drawable.ch));
            } else {
                Toast.makeText(getApplicationContext(), "Текущая неделя: Знаменатель", Toast.LENGTH_SHORT).show();
                menu.getItem(0).setIcon(getResources().getDrawable(R.drawable.ze));
        }
    }

    public void loadName() {
        // В качестве подзоголовка берем значение выбранной группы.
        typeGroupe = prefs.getString(getString(R.string.pref_groupe), "");
        // Для DB имена групп указаны английским алфавитом без пробелов и символов
        // Поэтому создаем второй массив на русском, сравниваем их по id

        String[] asu = getResources().getStringArray(R.array.groups_value);
        String[] names = getResources().getStringArray(R.array.all_group);

        int index = -1;

        for (int i = 0; (i < asu.length) && (index == -1); i++) {

            if (asu[i].equals(typeGroupe)) {
                index = i;
            }
        }
        // затем подставляем значение в качестве заголовка
        MainActivity.this.tb.setSubtitle(names[index]);

        // Если пользователь очистил данные в окне настроек, то строка будет пуста. Делаем проверку.
        if ((typeGroupe.length() == 0)) {
            startActivity(settingsIntent);
            finish();
        }
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {
        switch (item.getItemId()) {
            case R.id.main_tab:
                FragmentTransaction ft1 = FM.beginTransaction();
                ft1.replace(R.id.containerView, new TabFragment()).commit();
                loadName();
                break;
            case R.id.map:
                MainActivity.this.tb.setSubtitle("Учебные корпуса");
                FragmentTransaction ft2 = FM.beginTransaction();
                ft2.replace(R.id.containerView, new MapsFragment()).commit();
                break;
          case R.id.time_alarm:
                MainActivity.this.tb.setSubtitle("Время звонков");
                FragmentTransaction ft3 = FM.beginTransaction();
                ft3.replace(R.id.containerView, new TimeListFragment()).commit();
                break;
            case R.id.exzam:
                MainActivity.this.tb.setSubtitle("Экзаменационная сессия");
                FragmentTransaction ft4 = FM.beginTransaction();
                ft4.replace(R.id.containerView, new Session()).commit();
                break;
            default:
                break;
        }
        dl.closeDrawer(GravityCompat.START);
        return true;

    }
}

File: app/src/main/java/ru/vyaacheslav/suhov/imeit/activity/MainActivity.kt
package ru.vyaacheslav.suhov.imeit.activity

import android.content.Context
import android.content.Intent
import android.content.SharedPreferences
import android.net.Uri
import android.os.Build
import android.os.Bundle
import android.os.Handler
import android.preference.PreferenceManager
import android.support.v4.app.FragmentManager
import android.support.v4.app.FragmentTransaction
import android.support.v7.app.AppCompatActivity
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.view.animation.AnimationUtils
import kotlinx.android.synthetic.main.activity_main.*
import org.jetbrains.anko.intentFor
import ru.vyaacheslav.suhov.imeit.R
import ru.vyaacheslav.suhov.imeit.ftagments.bells.BellsFragment
import ru.vyaacheslav.suhov.imeit.ftagments.maps.MapsFragment
import ru.vyaacheslav.suhov.imeit.ftagments.schedule.ScheduleFragment
import ru.vyaacheslav.suhov.imeit.ftagments.schedule.TemporarySchedule

class MainActivity : AppCompatActivity() {

    private lateinit var ft: FragmentTransaction
    private lateinit var fm: FragmentManager
    private lateinit var prefs: SharedPreferences
    private lateinit var prefTh: String
    private lateinit var prefGr: String
    private var prefFab: Boolean = true
    private var prefWeek: Boolean = true

    private lateinit var settingsIntent: Intent

    val schedule: String = "https://yadi.sk/d/SRZmnHWF3aabGA"

    override fun onCreate(savedInstanceState: Bundle?) {

        prefs = PreferenceManager.getDefaultSharedPreferences(applicationContext)
        prefTh = prefs.getString(getString(R.string.pref_key_theme), "")
        prefGr = prefs.getString(getString(R.string.pref_key_group), "")
        prefFab = prefs.getBoolean(getString(R.string.pref_key_check_fab), true)
        prefWeek = prefs.getBoolean(getString(R.string.pref_key_check_week), true)

        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
            if (prefTh == "Светлая") {
                setTheme(R.style.ThemeWrithe)
            } else {
                setTheme(R.style.ThemeDark)
            }
        }

        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        // Интент для запуска окна с настройками
        val prefs = getPreferences(Context.MODE_PRIVATE)
        if (prefs.getBoolean("isFirstRun", true)) startActivity(intentFor<SettingsActivity>())
        prefs.edit().putBoolean("isFirstRun", false).apply()

        isWeek()
        // Стандартная инициализация компонетов
        setSupportActionBar(toolbar)

        // Инициализация транзакции фрагментов
        fm = supportFragmentManager
        ft = fm.beginTransaction()
        isGroupConnected() // По умолчанию - расписание.

        // Обработка нажатий на пункты меню.
        settingsIntent = Intent(this, SettingsActivity::class.java)
        groupName("Расписание") // Загрузка имени группы согласно настройкам;

        visibleFab(fab)
        bottom_navigation.selectedItemId = R.id.schedule
        bottom_navigation.setOnNavigationItemSelectedListener { item ->
            when (item.itemId) {
                R.id.times -> {
                    groupName("Время звонков")
                    val ft3 = fm.beginTransaction()
                    ft3.replace(R.id.container, BellsFragment()).commit()

           /*         fab.setImageResource(R.drawable.ic_assessment)
                    visibleFab(fab)
                    fab.setOnClickListener {
                        // Анализ расписания
                    }*/
                    invisibleView(fab)
                }
                R.id.schedule -> {
                    isGroupConnected()
                    groupName("Расписание")
                    fab.setImageResource(R.drawable.ic_download)
                    visibleFab(fab)
                    fab.setOnClickListener {
                        startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(schedule)))
                    }
                }
                R.id.location -> {
                    val ft2 = fm.beginTransaction()
                    ft2.replace(R.id.container, MapsFragment()).commit()
                    groupName("Корпуса")
                    invisibleView(fab)
                }
            }
            true
        }
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        return when (item.itemId) {
            R.id.settings -> {
                startActivity(settingsIntent)
                finish()
                super.onOptionsItemSelected(item)
            }
            else -> super.onOptionsItemSelected(item)
        }
    }

    private fun isGroupConnected() {

        val ft1 = fm.beginTransaction()
        if (prefGr == "isNot") {
            ft1.replace(R.id.container, TemporarySchedule()).commit()
        } else {
            ft1.replace(R.id.container, ScheduleFragment()).commit()
        }
    }

    private fun groupName(s: String?) {
        prefGr = prefs.getString(getString(R.string.pref_key_group), "")

        /*  val name = DB(this).groups()*/

        /* var i = 0
         while (i < name.size) {
             val str: String = name[i]["value"].toString() // HashMap  со всеми группами
             i++

             if (prefGr == "Нет группы") {
                 this.tb.subtitle = s
             } else {
                 this.tb.subtitle = "$str | $s"
             }
         }*/

        // В качестве подзоголовка берем значение выбранной группы.
        prefGr = prefs.getString(getString(R.string.pref_key_group), "")
        // Для DB имена групп указаны английским алфавитом без пробелов и символов
        // Поэтому создаем второй массив на русском, сравниваем их по id
        val asu = resources.getStringArray(R.array.groups_value)
        val names = resources.getStringArray(R.array.all_group)
        var index = -1
        var i = 0
        while (i < asu.size && index == -1) {
            if (asu[i] == prefGr) {
                index = i
            }
            i++
        }
        // затем подставляем значение в качестве заголовка
        this.toolbar.subtitle = names[index] + " | $s"
    }

    private fun isWeek() {
        if (prefWeek) {
            week.visibility = View.VISIBLE
            week.text = "Числитель"

            week.startAnimation(AnimationUtils.loadAnimation(this, R.anim.falling))
            container.startAnimation(AnimationUtils.loadAnimation(this, R.anim.bottom))
            Handler().postDelayed({
                invisibleView(week)
                container.startAnimation(AnimationUtils.loadAnimation(this, R.anim.top))
            }, 3000)
        } else
            week.visibility = View.GONE
    }

    private fun visibleFab(view: View) {
        if (prefFab) {
                view.visibility = View.VISIBLE
                view.startAnimation(AnimationUtils.loadAnimation(applicationContext, R.anim.view_alpha_visable))
        } else view.visibility = View.GONE
    }

    private fun invisibleView(view: View) {

        if (prefFab) {
            if (view.visibility != View.GONE) {
                view.startAnimation( AnimationUtils.loadAnimation(applicationContext, R.anim.view_alpha_gone))
                view.visibility = View.GONE
            }
        } else view.visibility = View.GONE

    }
}